starter-code/index.js

```diff
@@ -1,17 +1,66 @@
 class SortedList {
-  constructor() {}
+  constructor() {
+    this.items = [];
+    this.length = this.items.length;
+  }
 
-  add(item) {}
+  add(item) {
+    this.items.push(item);
+    this.length = this.items.length;
+    this.items.sort(function(a, b) {
+      return a - b;
+    });
+  }
 
-  get(pos) {}
+  get(pos) {
+    if (pos > this.items.length) {
+      throw new Error("OutOfBounds");
+    } else return this.items[pos];
+  }
 
-  max() {}
+  max() {
+    if (this.items.length == 0) {
+      throw new Error("EmptySortedList");
+    } else {
+      this.items.sort(function(a, b) {
+        return a - b;
+      });
 
-  min() {}
+      return this.items[this.items.length - 1];
+    }
+  }
 
-  sum() {}
+  min() {
+    if (this.items.length == 0) {
+      throw new Error("EmptySortedList");
+    } else {
+      this.items.sort(function(a, b) {
+        return b - a;
+      });
 
-  avg() {}
+      return this.items[this.items.length - 1];
+    }
+  }
+
+  sum() {
+    if (this.items.length == 0) {
+      return 0;
+    } else {
+      let result = this.items.reduce((a, b) => a + b, 0);
+
+      return result;
+    }
+  }
+
+  avg() {
+    if (this.items.length == 0) {
+      throw new Error("EmptySortedList");
+    } else {
+      let result = this.items.reduce((a, b) => a + b, 0) / this.items.length;
```

Comment:
Keep in mind that you can use the function: this.sum() so you don't have to reduce twice